darwin/Classes/WireguardDartPlugin.swift

```diff
@@ -1,109 +1,202 @@
+            guard let bundleId = args["bundleId"] as? String, !bundleId.isEmpty
+            else {
+                result(
+                    nativeFlutterError(message: "required argument: 'bundleId'")
+                )
                 return
             }
-            guard let tunnelName = args["tunnelName"] as? String, !tunnelName.isEmpty else {
-                result(nativeFlutterError(message: "required argument: 'tunnelName'"))
+            guard let tunnelName = args["tunnelName"] as? String,
+                !tunnelName.isEmpty
+            else {
+                result(
+                    nativeFlutterError(
+                        message: "required argument: 'tunnelName'"))
                 return
             }
-            Logger.main.debug("Tunnel bundle ID: \(bundleId), name: \(tunnelName)")
+            Logger.main.debug(
+                "Tunnel bundle ID: \(bundleId), name: \(tunnelName)")
             Task {
                 do {
-                    vpnManager = try await setupProviderManager(bundleId: bundleId, tunnelName: tunnelName)
+                    vpnManager = try await setupProviderManager(
+                        bundleId: bundleId, tunnelName: tunnelName)
                     Logger.main.debug("Tunnel setup OK")
                     result("")
                 } catch {
                     Logger.main.error("Tunnel setup ERROR: \(error)")
-                    result(nativeFlutterError(message: "could not setup VPN tunnel: \(error)"))
+                    result(
+                        nativeFlutterError(
+                            message: "could not setup VPN tunnel: \(error)"))
                     return
                 }
             }
         case "connect":
             Logger.main.debug("handle connect")
             let cfg: String
             if let args = call.arguments as? [String: Any],
-               let argCfg = args["cfg"] as? String {
+                let argCfg = args["cfg"] as? String
+            {
                 cfg = argCfg
             } else {
                 Logger.main.error("Required argument 'cfg' not provided")
                 result(nativeFlutterError(message: "required argument: 'cfg'"))
                 return
             }
             guard let mgr = vpnManager else {
-                Logger.main.error("Tunnel not initialized, missing 'vpnManager'")
-                result(nativeFlutterError(message: "tunnel not initialized, missing 'vpnManager'"))
+                Logger.main.error(
+                    "Tunnel not initialized, missing 'vpnManager'")
+                result(
+                    nativeFlutterError(
+                        message: "tunnel not initialized, missing 'vpnManager'")
+                )
                 return
             }
             Logger.main.debug("Connection configuration: \(cfg)")
             Task {
                 do {
+                    mgr.isEnabled = true
```

Comment:
What is the point to do check before connect if we have separate methods to check before it?

Reply:
In case the tunnel had gotten disabled, re-enable and save it, then connect.

Reply:
But it should as the pseudo code on application side:

```
if ! checkTunnelConfiguration() {
   setupTunnel()
}

connect()
```